### DISCUSSION TITLE
Divide by zero error

### DISCUSSION THREAD

**Discussion by chrimerss** [NASA-LIS/LISF#1448] [Q&A] [answered]
Hello,

I got the system error - floating divide by zero and was wondering if any of you have some insights. I was able to pin down the location of the error by printing the variables. I found my soil depths were all initialized to zeros for the first grid cell. Attached are my config files and lis output. 

Appreciate your help!

Allen

[ldt.txt](https://github.com/NASA-LIS/LISF/files/13342768/ldt.txt)
[lis.txt](https://github.com/NASA-LIS/LISF/files/13342769/lis.txt)
[lislog.txt](https://github.com/NASA-LIS/LISF/files/13342770/lislog.txt)
<img width="743" alt="Screenshot 2023-11-13 at 2 43 37 PM" src="https://github.com/NASA-LIS/LISF/assets/31950869/2b229018-84c1-4e52-86ae-cab513e6b876">

**Comment by dmocko** [comment]
Hi @chrimerss 

From where did you obtain the parameters in?: `PARAMS/CONUS_1km_HYMAP`

Can you please also attach your ldt/make/configure.ldt and lis/make/configure.lis files?

I am wondering if you configured LDT to use little-endian binary files; however, the files you are using are big-endian.

**Reply by agetiran** [reply]
Hi @chrimerss
Something is off here. the lis.config file shows that HyMAP is running in offline mode using NLDAS runoff as forcing:
Routing model:                  "HYMAP2 router"
External runoff data source: "NLDAS2 runoff data"
HYMAP2 routing model time step:              "30mn"
TEMPLATE model timestep:                     "30mn"

And the LSM is set as template:
Number of surface model types:           2
Surface model types:			"LSM" "Openwater"
Surface model output interval:	        "1da" 
Land surface model:			"none"
TEMPLATE model timestep:    1hr
Open water model:			"template open water"
 
It's not clear to me why the model is crashing in NoahMP401. Could you please double check the config files used in this simulation?

**Reply by chrimerss** [reply]
@agetiran -
Sorry for the confusion. I've tried many ways such as online mode and offline mode and forgot to update the config file. Attached are the latest ones. At this time, I got seg error.


@dmocko - 
I can confirm I configured using big-endian and it works perfectly when I crop a small domain instead of using the whole state.

<img width="919" alt="Screenshot 2023-11-14 at 9 46 35 PM" src="https://github.com/NASA-LIS/LISF/assets/31950869/e577496c-2ce6-4c62-aa61-70cb8d741f09">


[ldt.txt](https://github.com/NASA-LIS/LISF/files/13360662/ldt.txt)
[lisconfig.txt](https://github.com/NASA-LIS/LISF/files/13360663/lisconfig.txt)
[lislog.txt](https://github.com/NASA-LIS/LISF/files/13360664/lislog.txt)

**Comment by agetiran** [comment]
Hi @chrimerss. The error is likely related to inconsistencies in the parameters you are using. @dmocko previously asked where you obtained the HyMAP parameters at 1km. Would you mind responding to that question?

Also:
In your config file, NoahMP and HyMAP have different time steps
HYMAP2 routing model time step:              "30mn"
Noah-MP.4.0.1 model timestep:                "1hr"
They must be the same. I suggest using 15mn for both.

I suggest turning off the ensemble mode in HyMAP
HYMAP2 run in ensemble mode:                 0

**Reply by chrimerss** [reply]
Hi @agetiran @dmocko,

@agetiran you sent me the CONUS 1km parameter about a year ago.

I tried setting both timesteps to 15 mn but still didn't work. Same error.

**Reply by agetiran** [reply, marked as answer]
Just for a closure to this thread:
I worked offline with @chrimerss on the generation of a new customized parameter set for the domain, and that resolved the issue.
